src/commands/numbersCommands.ts

```diff
@@ -20,13 +23,36 @@ export class NumbersCommands {
     this.vonageNumbersViewDataProvider.refresh();
   };
 
-  filterNumbers = () => {
-    this.telemetry.sendEvent('number.search');
-    this.vonageNumbersViewDataProvider.searchNumbers();
-  }
-
   buyNumber = () => {
     this.telemetry.sendEvent('number.buy');
     this.vonageNumbersViewDataProvider.buyNumber();
   }
+
+  cancelNumber = (node?: NumberTreeItem) => {
+    this.telemetry.sendEvent('number.cancel');
+    if (node) {
+      this.vonageNumbersViewDataProvider.cancelNumber(node);
+    }
+  }
+
+  assignNumber = (node?: NumberTreeItem) => {
+    this.telemetry.sendEvent('number.assign');
+    if (node) {
+      this.vonageNumbersViewDataProvider.assignNumber(node);
+    }
+  }
+
+  unassignNumber = (node?: NumberTreeItem) => {
+    this.telemetry.sendEvent('number.unassign');
+    if (node) {
+      this.vonageNumbersViewDataProvider.unassignNumber(node);
+    }
+  }
+
+  copyNumber = (node?: NumberTreeItem) => {
+    this.telemetry.sendEvent('number.copy');
+    if (node) {
+      this.vonageNumbersViewDataProvider.copyNumber(node);
```

Comment:
copied value needs to be in the accepted format without the +.
see how copy works on dashboard (https://dashboard.nexmo.com/your-numbers)